File: app/src/main/java/com/sumian/hw/widget/SwitchDateView.java
package com.sumian.hw.widget;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import com.sumian.sd.R;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

/**
 * Created by sm
 * on 2018/3/6.
 * desc:
 */

public class SwitchDateView extends FrameLayout implements View.OnClickListener {

    ImageView mIvPre;
    TextView mTvDate;
    ImageView mIvNext;

    private long mUnixTime;

    private OnSwitchDateListener mOnSwitchDateListener;

    public SwitchDateView(Context context) {
        this(context, null);
    }

    public SwitchDateView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public SwitchDateView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setBackgroundColor(getResources().getColor(R.color.pick_bg_color));
        initView(context);
        this.mTvDate.setText(new SimpleDateFormat("yyyy/MM/dd", Locale.getDefault()).format(new Date()));
    }

    private void initView(Context context) {
        View inflate = inflate(context, R.layout.hw_lay_switch_date_view, this);
        mIvPre = inflate.findViewById(R.id.iv_pre);
        mTvDate = inflate.findViewById(R.id.tv_date);
        mIvNext = inflate.findViewById(R.id.iv_next);
        inflate.findViewById(R.id.iv_pre).setOnClickListener(this);
        inflate.findViewById(R.id.iv_next).setOnClickListener(this);
    }

    public void setOnSwitchDateListener(OnSwitchDateListener onSwitchDateListener) {
        mOnSwitchDateListener = onSwitchDateListener;
    }


    @Override
    public void onClick(View v) {
        int i = v.getId();
        if (i == R.id.iv_pre) {
            if (mOnSwitchDateListener != null) {
                mOnSwitchDateListener.scrollToTime(mUnixTime - 60 * 60 * 24);
            }
        } else if (i == R.id.iv_next) {
            if (mOnSwitchDateListener != null) {
                mOnSwitchDateListener.scrollToTime(mUnixTime + 60 * 60 * 24);
            }
        }
    }

    public long getTodayUnixTime() {
        Calendar instance = Calendar.getInstance();
        int year = instance.get(Calendar.YEAR);
        int month = instance.get(Calendar.MONTH);
        int date = instance.get(Calendar.DATE);
        instance.set(year, month, date, 0, 0, 0);
        return instance.getTimeInMillis() / 1000L;
    }

    public long getUnixTime() {
        return mUnixTime;
    }

    public void setUnixTime(long unixTime) {
        this.mUnixTime = unixTime;
        if (getTodayUnixTime() == unixTime) {
            mIvNext.setVisibility(INVISIBLE);
        } else {
            mIvNext.setVisibility(VISIBLE);
        }
        this.mTvDate.setText(new SimpleDateFormat("yyyy/MM/dd", Locale.getDefault()).format(new Date(unixTime * 1000L)));
    }

    public interface OnSwitchDateListener {

        void scrollToTime(long unixTime);

    }
}

File: app/src/main/java/com/sumian/hw/report/widget/SwitchDateView.kt
package com.sumian.hw.report.widget

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import com.sumian.sd.R
import kotlinx.android.synthetic.main.hw_lay_switch_date_view.view.*
import java.text.SimpleDateFormat
import java.util.*

@Suppress("DEPRECATION")
/**
 * Created by sm
 * on 2018/3/6.
 * desc:
 */

class SwitchDateView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
        FrameLayout(context, attrs, defStyleAttr), View.OnClickListener {

    var unixTime: Long = 0
        set(unixTime) {
            field = unixTime
            if (todayUnixTime == unixTime) {
                iv_next.visibility = View.INVISIBLE
            } else {
                iv_next.visibility = View.VISIBLE
            }
            this.tv_date.text = SimpleDateFormat("yyyy/MM/dd", Locale.getDefault()).format(Date(unixTime * 1000L))
        }

    private var mOnSwitchDateListener: OnSwitchDateListener? = null

    val todayUnixTime: Long
        get() {
            val instance = Calendar.getInstance()
            val year = instance.get(Calendar.YEAR)
            val month = instance.get(Calendar.MONTH)
            val date = instance.get(Calendar.DATE)
            instance.set(year, month, date, 0, 0, 0)
            return instance.timeInMillis / 1000L
        }

    init {
        initView(context)
        tv_date.text = SimpleDateFormat("yyyy/MM/dd", Locale.getDefault()).format(Date())
    }

    private fun initView(context: Context) {
        View.inflate(context, R.layout.hw_lay_switch_date_view, this)
        iv_pre.setOnClickListener(this)
        iv_next.setOnClickListener(this)
    }

    fun setOnSwitchDateListener(onSwitchDateListener: OnSwitchDateListener): SwitchDateView {
        mOnSwitchDateListener = onSwitchDateListener
        return this
    }


    override fun onClick(v: View) {
        when (v.id) {
            R.id.iv_pre -> {
                mOnSwitchDateListener?.onScrollToTime(unixTime - 60 * 60 * 24)
            }
            R.id.iv_next -> {
                mOnSwitchDateListener?.onScrollToTime(unixTime + 60 * 60 * 24)
            }
        }
    }

    interface OnSwitchDateListener {

        fun onScrollToTime(unixTime: Long)

    }
}
